specs/WebMessageObjects.md

```diff
@@ -227,148 +525,37 @@ interface WebView extends EventTarget {
-            // information about sending messages from the HTML document in the WebView to the
-            // host, navigate to <see cref="CoreWebView2.WebMessageReceived"/>. The message is sent
-            // asynchronously. If a navigation occurs before the message is posted to the page,
-            // the message is not be sent.
-            // </remarks>
-            // <example>
-            // Runs the message event of the <c>window.chrome.webview</c> of the top-level
-            // document. JavaScript in that document may subscribe and unsubscribe to the event
-            // using the following code:
-            // $net$
-            // <code>
-            // window.chrome.webview.addEventListener('message', handler)
-            // window.chrome.webview.removeEventListener('message', handler)
-            // </code>
-            // $end$
-            // $winrt$
-            // ```javascript
-            // window.chrome.webview.addEventListener('message', handler)
-            // window.chrome.webview.removeEventListener('message', handler)
-            // ```
-            // $end$
-            // </example>
-            // <seealso cref="CoreWebView2Settings.IsWebMessageEnabled"/>
-            // <seealso cref="WebMessageReceived"/>
-            // <seealso cref="PostWebMessageAsString"/>
-            void PostWebMessageAsJsonWithAdditionalObjects(
-                String webMessageAsJson,
-                IVectorView<Object> additionalObjects);
-        }
-    }
-}
-```
+interface WebViewMessageEvent extends MessageEvent {
+    /**
+     * The source of the event is the `chrome.webview` object.
+     */
+    source: WebView;
 
-```ts
-interface WebViewMessageEvent: MessageEvent {
-  /* When a WebMessage is posted via PostWebMessageAsJsonWithAdditionalObjects
-   * this property will contain the DOM objects that are injected to the content.
-   * Currently the following types are supported:
-   * - File
-   */
-  additionalObjects: ArrayLike<any>
+    /**
+     * Additional DOM objects that are sent via PostJSONMessageWithAdditionalObjects.
+     * These objects can be of following types:
+     * | DOM       | Win32       | .NET/WinRT |
+     * |-----------|-------------|------------|
+     * | [File](https://developer.mozilla.org/docs/Web/API/File] | ICoreWebView2File | CoreWebView2File |
+     * | [FileSystemHandle](https://developer.mozilla.org/docs/Web/API/FileSystemHandle) | ICoreWebView2FileSystemHandle | CoreWebView2FileSystemHandle |
+     * | null      | nullptr     | null       |
+     * | undefined | nullptr     | null       |
+     */
+    additionalObjects: ArrayLike<any>
 }
 
+// Matching `interface MessagePort` from lib.dom.d.ts for messaging APIs
+/**
+ * Events of the `WebView` interface.
+ */
 interface WebViewEventMap {
     "message": WebViewMessageEvent;
-    ...
+    "sharedbufferreceived": SharedBufferReceivedEvent;

```

Comment:
Not sure why this is here in this spec.

Reply:
This is unrelated to this feature. Please go back to '...'